src/btree/row_key.c

```diff
@@ -496,6 +500,10 @@ __wt_row_ikey(WT_SESSION_IMPL *session,
 	ikey->cell_offset = cell_offset;
 	memcpy(WT_IKEY_DATA(ikey), key, size);
 
-	*(WT_IKEY **)ikeyp = ikey;
+	oldv = (uintptr_t)*ikeyp;
+	WT_DIAGNOSTIC_YIELD;
+	WT_ASSERT(session, oldv == 0 || (oldv & WT_IK_FLAG) != 0);
+	success = WT_ATOMIC_CAS8(*ikeyp, (WT_IKEY *)oldv, ikey);
+	WT_ASSERT(session, success);
```

Comment:
If the CAS and swap fails, we'll end up returning 0 and not switching in the new key. Should we have completely different code paths here, where the non-diagnostic version works like the old version?